### PR TITLE
feat: support custom admin version

### LINKED ISSUE #240: Add support for developers to set admin version
add a way to configure an admin version the sidekick should use when requesting the admin.
this can be useful to test new admin features with the sidekick.

the admin version can be selected with the `hlx-admin-version` query parameter.

### PR DESCRIPTION
**Pull request by rofe**
Fix #240

### CODE REVIEW

File: src/sidekick/module.js [hunk 1]
```diff
@@ -584,15 +585,21 @@
    * @param {string} path The current path
    * @returns {string} The admin URL
    */
-  function getAdminUrl({ owner, repo, ref }, api, path) {
-    return new URL([
+  function getAdminUrl({
```
Comment (tripodsan) [inline src/sidekick/module.js:588]: if you would return a `Url` instead of a string, you could manipulate it better later....

Comment (rofe) [reply src/sidekick/module.js:588]: Actually, I do return a `URL` - the jsdoc was wrong 😄

File: src/sidekick/module.js [hunk 2]
```diff
@@ -1212,9 +1219,9 @@
           'status',
           this.isEditor() ? '' : pathname,
         );
-        apiUrl.search = new URLSearchParams([
-          ['editUrl', this.isEditor() ? href : 'auto'],
-        ]).toString();
+        const params = new URLSearchParams(apiUrl.search);
+        params.set('editUrl', this.isEditor() ? href : 'auto');
+        apiUrl.search = params.toString();
```
Comment (tripodsan) [inline src/sidekick/module.js:1224]:
```suggestion
        apiUrl.searchParams.append('editUrl', this.isEditor() ? href : 'auto');
```

**Review by tripodsan — APPROVED** [review]

**Pull request merged**